src/py/flwr/cli/run/run.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower command line interface `new` command."""
+
+import sys
+
+import typer
+
+from flwr.cli.flower_toml import apply_defaults, load_flower_toml, validate_flower_toml
+from flwr.simulation.run_simulation import _run_simulation
+
+
+def run() -> None:
+    """Run Flower project."""
+    config = load_flower_toml()
+    if not config:
+        print(
+            typer.style(
+                "Project configuration could not be loaded. "
+                "flower.toml does not exist.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+        )
+        sys.exit()
+
+    is_valid, reasons = validate_flower_toml(config)
+    if not is_valid:
+        print(
+            typer.style(
+                "Project configuration could not be loaded.\nflower.toml is invalid:\n"
+                + "\n".join([f"- {line}" for line in reasons]),
+                fg=typer.colors.RED,
+                bold=True,
+            )
+        )
+
+    # Apply defaults
+    defaults = {
+        "flower": {
+            "engine": {"name": "simulation", "simulation": {"super-node": {"num": 100}}}
+        }
+    }
+    config = apply_defaults(config, defaults)
+
+    server_app_ref = config["flower"]["components"]["serverapp"]
+    client_app_ref = config["flower"]["components"]["clientapp"]
+    engine = config["flower"]["engine"]["name"]
+
+    if engine == "simulation":
+        num_supernodes = config["flower"]["engine"]["simulation"]["super-node"]["num"]
+
+        _run_simulation(
```

Comment:
@jafermarq should this use `_run_simulation` or `run_simulation` or sth else?

One alternative would be to decouple `flwr run` and `flower-simulation` would be to create a subprocess here that calls `flower-simulation` with appropriate arguments.

Reply:
Calling `_run_simulation()` is correct. What's done in this `run()` closely follows how [`run_simulation_from_cli()`](https://github.com/adap/flower/blob/a5cf7359997a634cd8ae46de607275301a79b93c/src/py/flwr/simulation/run_simulation.py#L44) is intended to be used (which just does argparsing and then call `_run_simulation()`)